test/pkg/require/requires.go

```diff
@@ -11,7 +11,7 @@ import (
 // Features checks the wanted features are present in given,
 // and skips the test if any are missing. If the given set is empty,
 // the function returns (by default all features are enabled).
-func Features(tb testing.TB, given *flag.StringSet, want ...string) {
+func Features(tb testing.TB, given *flag.ExcludeStringSet, want ...string) {
```

Comment:
>given *flag.ExcludeStringSet [](http://example.com/codeflow?start=29&length=28)

Not sure how/where the -exclude flag is parsed, but now that given contains the ExcludeStringSet, the logic for line 19-21 should probably be updated.

Reply:
`ExcludeStringSet` also defines an `IsSet` function, like `StringSet`, which includes the `-exclude` flag logic, so that change is opaque to `Features`.
`NewExcludeStringSet` defines (via go's built in `"flag"` pkg) the command line flags `-features` and `-exclude`, so wraps all the necessary logic.

Reply:
O I get it now.